File: app/src/main/java/com/itwiggle/randomchime/AlarmReceiver.java
package com.itwiggle.randomchime;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class AlarmReceiver extends BroadcastReceiver {
    @Override public void onReceive(Context c, Intent i) {
        Intent service=new Intent(c,PlaybackService.class); service.setAction("PLAY"); c.startForegroundService(service);
        Store.addHistory(c,"prompted");
    }
}

File: app/src/main/java/com/itwiggle/randomchime/AlarmReceiver.kt
package com.itwiggle.randomchime

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class AlarmReceiver:BroadcastReceiver(){override fun onReceive(context:Context,intent:Intent){val prefs=AppPrefs(context);val scheduledAt=prefs.nextAt();prefs.markFired();context.startForegroundService(Intent(context,PromptPlaybackService::class.java).setAction(PromptPlaybackService.PLAY).putExtra("scheduled_at",scheduledAt));AlarmScheduler.scheduleNext(context)}}
